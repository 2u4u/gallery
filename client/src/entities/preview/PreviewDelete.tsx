import { PreviewType } from 'shared/types/index.type';

type Props = {
    preview: PreviewType;
    handleDelete: (file: File) => void;
}

function PreviewDelete({ preview, handleDelete }: Props) {
    return (preview.downloaded !== undefined ? null :  
        (
            <button onClick={() => handleDelete(preview.file)}>
                <img src="./img/close.svg" alt="close modal" className='preview__close'/>
            </button>
        )
    );
}

export default PreviewDelete;
