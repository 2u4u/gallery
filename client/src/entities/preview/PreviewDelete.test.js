import renderer from 'react-test-renderer';
import PreviewDelete from './PreviewDelete';

describe('Preview delete component ', () => {
    const preview = {
        type: 'image',
        url: 'https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png',
        name: 'test image',
        file: 'Fake file',
    };
    const handleDelete = () => false;
    it('matches snapshot', () => {
        const component = renderer.create(
            <PreviewDelete handleDelete={handleDelete} preview={preview} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
