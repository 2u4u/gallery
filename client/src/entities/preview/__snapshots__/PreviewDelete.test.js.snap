// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview delete component  matches snapshot 1`] = `
<button
  onClick={[Function]}
>
  <img
    alt="close modal"
    className="preview__close"
    src="./img/close.svg"
  />
</button>
`;
